noarkexchange/src/main/java/no/difi/meldingsutveksling/noarkexchange/PayloadUtil.java

```diff
@@ -98,6 +108,54 @@ public static String queryPayload(Object payload, String xpath) throws PayloadEx
         return result;
     }
 
+    public static List<NoarkDocument> parsePayloadForDocuments(Object payload) throws PayloadException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "parsePayloadForDocuments" is 12 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)